sweep/sweeper.go

```
@@ -1,40 +1,701 @@
+				continue
+			}
+
+			// Signal sweep results for inputs in this confirmed tx.
+			for _, txIn := range spend.SpendingTx.TxIn {
+				outpoint := txIn.PreviousOutPoint
+
+				// Check if this input is known to us. It could
+				// probably be unknown if we canceled the
+				// registration, deleted from pendingInputs but
+				// the ntfn was in-flight already. Or this could
+				// be not one of our inputs.
+				_, ok := s.pendingInputs[outpoint]
+				if !ok {
+					continue
+				}
+
+				// Return either a nil or a remote spend result.
+				var err error
+				if !isOurTx {
+					err = ErrRemoteSpend
+				}
+
+				// Signal result channels.
+				s.signalAndRemove(&outpoint, Result{
+					Tx:  spend.SpendingTx,
+					Err: err,
+				})
+			}
+
+			// Now that an input of ours is spent, we can try to
+			// resweep the remaining inputs.
+			s.scheduleSweep(bestHeight)
+
+		// The timer expires and we are going to (re)sweep.
+		case <-s.timer:
+			log.Debugf("Timer expired")
+
+			// Set timer to nil so we know that a new timer needs to
+			// be started when new inputs arrive.
+			s.timer = nil
+
+			// Retrieve fee estimate for input filtering and final
+			// tx fee calculation.
+			satPerKW, err := s.cfg.Estimator.EstimateFeePerKW(
+				s.cfg.SweepTxConfTarget,
+			)
+			if err != nil {
+				return err
+			}
+
+			// Examine pending inputs and try to construct lists of
+			// inputs.
+			inputLists, err := s.getInputLists(bestHeight, satPerKW)
+			if err != nil {
+				log.Errorf("Cannot sweep on timer event: %v",
+					err)
+			}
+
+			// Sweep selected inputs.
+			for _, inputs := range inputLists {
+				err := s.sweep(inputs, satPerKW, bestHeight)
+				if err != nil {
+					return err
+				}
+			}
+
+		// A new block comes in. Things may have changed, so we retry a
+		// sweep.
+		case epoch, ok := <-blockEpochs:
+			if !ok {
+				return nil
+			}
+
+			bestHeight = epoch.Height
+			log.Debugf("Block %v received", bestHeight)
+
+			s.scheduleSweep(bestHeight)
```

Also error is ignored.

---

Fixed in later commit.